purchases/src/main/kotlin/com/revenuecat/purchases/deeplinks/RCBillingPurchaseRedemptionHelper.kt

```diff
@@ -0,0 +1,60 @@
+package com.revenuecat.purchases.deeplinks
+
+import com.revenuecat.purchases.CacheFetchPolicy
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.CustomerInfoHelper
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.common.Backend
+import com.revenuecat.purchases.common.debugLog
+import com.revenuecat.purchases.common.errorLog
+import com.revenuecat.purchases.deeplinks.DeepLinkHandler.DeepLink
+import com.revenuecat.purchases.identity.IdentityManager
+import com.revenuecat.purchases.interfaces.ReceiveCustomerInfoCallback
+import com.revenuecat.purchases.interfaces.RedeemRCBillingPurchaseListener
+import com.revenuecat.purchases.interfaces.RedeemRCBillingPurchaseResult
+
+internal class RCBillingPurchaseRedemptionHelper(
+    private val backend: Backend,
+    private val identityManager: IdentityManager,
+    private val customerInfoHelper: CustomerInfoHelper,
+) {
+    var redeemRCBillingPurchaseListener: RedeemRCBillingPurchaseListener? = null
+
+    fun handleRedeemRCBPurchase(deepLink: DeepLink.RedeemRCBPurchase) {
+        debugLog("Detected RCB purchase redemption. Asking callback to initiate redeem.")
+        redeemRCBillingPurchaseListener?.handleRCBillingPurchaseRedemption { resultListener ->
```

Comment:
Right now, it's mandatory to set the listener in order to process the redemption link or it will be ignored.